src/wallet/walletdb.cpp

```diff
@@ -511,6 +541,23 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
 
             wss.nZKeys++;
         }
+        else if (strType == "sapzkey")
+        {
+            libzcash::SaplingPaymentAddress addr;
+            ssKey >> addr;
+            libzcash::SaplingExtendedSpendingKey key;
+            ssValue >> key;
+
+            if (!pwallet->LoadSaplingZKey(key, addr))
```

Comment:
Argh, this is an example where we won't know an address to insert here (because that data will be stored in a different record), so we *do* need the second parameter to be optional. Sorry @Eirik0 :( Hopefully we can just revert that commit without too much trouble.